### PR TITLE
treewide: update darwin sdk hashes

### LINKED ISSUE #201348: Darwin opensource packages are no longer available
###### Project name
This is in the file: `pkgs/os-specific/darwin/apple-source-releases/macos-11.0.1.nix`

###### Notify maintainers

maintainers: ???

###### Note for maintainers

`bootstrap_cmds`, for example, is given at version 121, but fails to download from every machine I've tried:
```
wget http://www.opensource.apple.com/tarballs/bootstrap_cmds/bootstrap_cmds-121.tar.gz
```

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Closes #201348

(But I'm not sure, since the hashes were updated manually...)

Feel free to edit the commit, especially if some of the hashes were wrong.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build stdenv

File: pkgs/os-specific/darwin/apple-source-releases/adv_cmds/boot.nix [hunk 1]
```diff
@@ -11,8 +11,8 @@
 # the more recent adv_cmds release is used for everything else in this package
 
 let recentAdvCmds = fetchzip {
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/apple-source-releases/adv_cmds/boot.nix:13]: Pelase use fetchFromGitHub

File: pkgs/tools/filesystems/hfsprogs/default.nix [hunk 2]
```diff
@@ -9,8 +9,8 @@ stdenv.mkDerivation rec {
       sha256 = "62d9b8599c66ebffbc57ce5d776e20b41341130d9b27341d63bda08460ebde7c";
     })
     (fetchurl {
-      url = "https://opensource.apple.com/tarballs/diskdev_cmds/diskdev_cmds-${version}.tar.gz";
-      sha256 = "74c9aeca899ed7f4bf155c65fc45bf0f250c0f6d57360ea953b1d536d9aa45e6";
+      url = "https://github.com/apple-oss-distributions/diskdev_cmds/archive/refs/tags/diskdev_cmds-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/hfsprogs/default.nix:12]: Same here

File: pkgs/tools/system/pdisk/default.nix [hunk 3]
```diff
@@ -13,8 +13,8 @@ stdenv.mkDerivation rec {
   version = "0.9";
 
   src = fetchzip {
-    url = "https://opensource.apple.com/tarballs/pdisk/pdisk-${lib.versions.minor version}.tar.gz";
-    sha256 = "sha256-+gBgnk/1juEHE0nXaz7laUaH7sxrX5SzsLGr0PHsdHs=";
+    url = "https://github.com/apple-oss-distributions/pdisk/archive/refs/tags/pdisk-${lib.versions.minor version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/tools/system/pdisk/default.nix:16]: and here

File: pkgs/os-specific/darwin/apple-source-releases/default.nix [hunk 4]
```diff
@@ -234,78 +234,78 @@ let
...
+    dyld            = applePackage "dyld"              "osx-10.12.6"     "sha256-JmKnOZtBPf96zEx7vhYHLBSTOPyKN71IdYE3R0IeJww=" {};
+    eap8021x        = applePackage "eap8021x"          "osx-10.11.6"     "sha256-54P3+YhVhOanoZQoqswDnr/GbR/AdEERse135nyuIQo=" {};
+    IOKit           = applePackage "IOKit"             "osx-10.11.6"     "" { inherit IOKitSrcs; };
+    launchd         = applePackage "launchd"           "osx-10.9.5"      "sha256-dmV0UK7hG9wvTr+F4Z47nCFXcVZCV+cQ46WbE0DBtJs=" {};
+    libauto         = applePackage "libauto"           "osx-10.9.5"      "sha256-GnRcKq8jRbEsI/PSDphwUjWtpEIEcnLlQL9yxYLgSsU=" {};
+    Libc            = applePackage "Libc"              "osx-10.12.6"     "sha256-LSsL7S3KFgGU9qjK4atu/4wBh8ftgfsk6JOvg+ZTZOY=" {
       Libc_10-9 = fetchzip {
-        url    = "http://www.opensource.apple.com/tarballs/Libc/Libc-997.90.3.tar.gz";
-        sha256 = "1xchgxkxg5288r2b9yfrqji2gsgdap92k4wx2dbjwslixws12pq7";
+        url    = "https://github.com/apple-oss-distributions/Libc/archive/refs/tags/Libc-997.90.3.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/apple-source-releases/default.nix:262]: and here

File: pkgs/os-specific/darwin/apple-source-releases/default.nix [hunk 5]
```diff
@@ -146,7 +146,7 @@ let
       then stdenv.fetchurlBoot
       else fetchurl;
   in fetch {
-    url = "http://www.opensource.apple.com/tarballs/${pname}/${pname}-${version}.tar.gz";
+    url = "https://github.com/apple-oss-distributions/${pname}/archive/refs/tags/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/apple-source-releases/default.nix:149]: and probably here, too

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
FYI due to cache misses ofborg will unlikely build anything for x86_64-darwin.

File: pkgs/os-specific/darwin/apple-source-releases/default.nix (hunk 5 above)
Comment (wegank) [reply pkgs/os-specific/darwin/apple-source-releases/default.nix:149]: I've fixed the others, but this one seems tricky.

**Comment by Artturin** [comment]
> > Is this blocked on anything? Are you trying to update the `fetch` to `fetchFromGithub`?
> 
> It was blocked by the changes in [community/community#46034](https://github.com/orgs/community/discussions/46034), which seems to be reverted now.

fetchFromGitHub and fetchzip hash the output of the archive not the archive itself so the change doesn't affect us

**Comment by wegank** [comment]
We're using `stdenv.fetchurlBoot` and `fetchurl` in this PR, unfortunately...

**Comment by winterqt** [comment]
It seems that Apple changed the locations of some headers, which breaks IOKit; see https://github.com/NixOS/nixpkgs/pull/215401#issuecomment-1427049425.

Please test things like this in the future, when possible (especially if OfBorg can't). If compute power is stopping you, see https://github.com/winterqt/darwin-build-box.

**Comment by winterqt** [comment]
Also: @Artturin, please don't disregard comments from a committer and go ahead and merge something, as they were obviously said for a reason. Thanks.

**Comment by Artturin** [comment]
> Also: @Artturin, please don't disregard comments from a committer and go ahead and merge something, as they were obviously said for a reason. Thanks.

I don't know what comment you're referring to

**Comment by winterqt** [comment]
> > Also: @Artturin, please don't disregard comments from a committer and go ahead and merge something, as they were obviously said for a reason. Thanks.
> 
> I don't know what comment you're referring to

Ah, apologies.

I meant https://github.com/NixOS/nixpkgs/pull/210449#issuecomment-1418688838, but I misunderstood and thought it was still blocked on it at the time.

Either way, your comment (https://github.com/NixOS/nixpkgs/pull/210449#issuecomment-1419045070) isn't true because of https://github.com/NixOS/nixpkgs/pull/210449#issuecomment-1419047794.

Sorry, though -- my apologies for misreading the dialogue here.

**Pull request merged**